src/background/stores/assessment-card-selection-store.ts

```diff
@@ -0,0 +1,219 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { AssessmentCardSelectionActions } from 'background/actions/assessment-card-selection-actions';
+import { TabActions } from 'background/actions/tab-actions';
+import { IndexedDBDataKeys } from 'background/IndexedDBDataKeys';
+import { PersistentStore } from 'common/flux/persistent-store';
+import { IndexedDBAPI } from 'common/indexedDB/indexedDB';
+import { Logger } from 'common/logging/logger';
+import { AssessmentCardSelectionStoreData } from 'common/types/store-data/assessment-card-selection-store-data';
+import { forOwn, isEmpty } from 'lodash';
+import { StoreNames } from '../../common/stores/store-names';
+import { RuleExpandCollapseData } from '../../common/types/store-data/card-selection-store-data';
+import {
+    CardSelectionPayload,
+    RuleExpandCollapsePayload,
+    UnifiedScanCompletedPayload,
+} from '../actions/action-payloads';
+import { UnifiedScanResultActions } from '../actions/unified-scan-result-actions';
+
+export class AssessmentCardSelectionStore extends PersistentStore<AssessmentCardSelectionStoreData> {
```

Comment:
We cannot use the exact same types as we do with CardSelectionStore. Specifcally we should create new payloads which should specify which assessment needs to be updated, using assessment keys (i.e. "automated-checks"). This is in relation to the comment specifying that this store contains card selection store data for all tests in an assessment.